File: app/src/main/java/com/example/app/MyModule.java
package com.example.app;

import com.example.app.MyExample;
import com.example.app.MyExampleImpl;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class MyModule {
    @Provides
    @Singleton
    static MyExample provideMyExample() {
        return new MyExampleImpl();
    }
}

File: app/src/main/java/com/example/app/MyModule.kt
package com.example.app

import com.example.app.MyExample
import com.example.app.MyExampleImpl

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class MyModule {
    @Provides
    @Singleton
    internal fun provideMyExample(): MyExample {
        return MyExampleImpl()
    }
}
